spec/text-editor-presenter-spec.coffee

```diff
@@ -657,140 +657,103 @@ describe "TextEditorPresenter", ->
           expectStateUpdate presenter, -> editor.setPlaceholderText("new-placeholder-text")
           expect(presenter.getState().content.placeholderText).toBe "new-placeholder-text"
 
-      describe ".lines", ->
-        lineStateForScreenRow = (presenter, screenRow) ->
-          presenter.getState().content.lines[presenter.model.tokenizedLineForScreenRow(screenRow).id]
+      describe ".tiles", ->
+        lineStateForScreenRow = (presenter, row) ->
+          lineId  = presenter.model.tokenizedLineForScreenRow(row).id
+          tileRow = presenter.tileForRow(row)
+          presenter.getState().content.tiles[tileRow].lines[lineId]
 
-        it "contains states for lines that are visible on screen, plus and minus the overdraw margin", ->
-          presenter = buildPresenter(explicitHeight: 15, scrollTop: 50, lineHeight: 10, lineOverdrawMargin: 1)
+        it "contains states for tiles that are visible on screen", ->
+          presenter = buildPresenter(explicitHeight: 6, scrollTop: 0, lineHeight: 1, tileCount: 3)
 
-          expect(lineStateForScreenRow(presenter, 3)).toBeUndefined()
-
-          line4 = editor.tokenizedLineForScreenRow(4)
-          expectValues lineStateForScreenRow(presenter, 4), {
-            screenRow: 4
-            text: line4.text
-            tokens: line4.tokens
-            top: 10 * 4
+          expectValues presenter.getState().content.tiles[0], {
+            top: 0
           }
-
-          line5 = editor.tokenizedLineForScreenRow(5)
-          expectValues lineStateForScreenRow(presenter, 5), {
-            screenRow: 5
-            text: line5.text
-            tokens: line5.tokens
-            top: 10 * 5
+          expectValues presenter.getState().content.tiles[2], {
+            top: 2
           }
-
-          line6 = editor.tokenizedLineForScreenRow(6)
-          expectValues lineStateForScreenRow(presenter, 6), {
-            screenRow: 6
-            text: line6.text
-            tokens: line6.tokens
-            top: 10 * 6
+          expectValues presenter.getState().content.tiles[4], {
+            top: 4
           }
-
-          line7 = editor.tokenizedLineForScreenRow(7)
-          expectValues lineStateForScreenRow(presenter, 7), {
-            screenRow: 7
-            text: line7.text
-            tokens: line7.tokens
-            top: 10 * 7
+          expectValues presenter.getState().content.tiles[6], {
+            top: 6
           }
 
-          line8 = editor.tokenizedLineForScreenRow(8)
-          expectValues lineStateForScreenRow(presenter, 8), {
-            screenRow: 8
-            text: line8.text
-            tokens: line8.tokens
-            top: 10 * 8
-          }
+          expect(presenter.getState().content.tiles[8]).toBeUndefined()
 
-          expect(lineStateForScreenRow(presenter, 9)).toBeUndefined()
-
-        it "does not overdraw above the first row", ->
```

Comment:
You're correct to drop it with tiling. It used to be required to prevent a full screen repaint when adding/removing lines at the top/bottom. If the repainted regions were too close together they would be grouped. This approach bypasses all such nonsense.